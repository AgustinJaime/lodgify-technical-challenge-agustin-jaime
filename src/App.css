/* Mobile first */
.app-container {
  width: 100%;
  background-color: var(--white);
}

.app-header {
  background-color: var(--main-color);
  padding: var(--p-small);
  text-transform: uppercase;
  text-align: right;
  border-top-left-radius: var(--s-small);
  border-top-right-radius: var(--s-small);
  box-shadow: 0 1px 1px 0 rgb(202, 202, 202);
}

@media screen and (min-width: 576px) {
  .app-container {
    max-width: var(--web-max-width);
    margin-top: var(--p-medium);
    border-radius: var(--s-small);
    width: 80%;
    box-shadow: 0px 2px 5px 1px rgb(202, 202, 202);
  }
}
