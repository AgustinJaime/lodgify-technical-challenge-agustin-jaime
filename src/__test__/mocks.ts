const fetchedDataMock = [
  {
    name: "Group 1",
    tasks: [
      {
        description: "Task 1 - 1",
        value: 10,
        checked: true,
      },
      {
        description: "Task 2 - 1",
        value: 20,
        checked: false,
      },
    ],
  },
  {
    name: "Group 2",
    tasks: [
      {
        description: "Task 1 - 1",
        value: 30,
        checked: false,
      },
      {
        description: "Task 2 - 1",
        value: 20,
        checked: false,
      },
    ],
  },
  {
    name: "Group 3",
    tasks: [
      {
        description: "Task 1 - 1",
        value: 20,
        checked: false,
      },
    ],
  },
];

export const mockInitialState = {
  tasks: {
    loading: "SUCCEEDED",
    data: fetchedDataMock,
  },
};
