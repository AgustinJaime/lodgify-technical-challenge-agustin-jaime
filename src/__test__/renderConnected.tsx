import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import reducers from "../store/reducers";

const renderConnected = (
  ui: React.ReactElement,
  initialState = {},
  {
    store = configureStore({
      reducer: reducers,
      preloadedState: initialState,
    }),
    ...renderOptions
  } = {}
) => {
  const Wrapper = ({ children }: React.PropsWithChildren) => (
    <Provider store={store}>{children}</Provider>
  );

  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export default renderConnected;
