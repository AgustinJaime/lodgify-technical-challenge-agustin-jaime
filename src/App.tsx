import "./App.css";
import ProgressBar from "./components/ProgressBar";

function App() {
  return (
    <div className="app-container">
      <header className="app-header">
        <h1>Lodgify Grouped Tasks</h1>
      </header>
      <ProgressBar />
    </div>
  );
}

export default App;
