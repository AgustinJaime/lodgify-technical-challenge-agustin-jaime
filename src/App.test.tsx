import App from "./App";
import { mockInitialState } from "./__test__/mocks";
import renderConnected from "./__test__/renderConnected";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => jest.fn(),
}));

describe("<App>", () => {
  it("should render without issues", () => {
    const documentBody = renderConnected(<App />, { ...mockInitialState });
    expect(documentBody).toMatchSnapshot();
  });
  it("should render header text", () => {
    const documentBody = renderConnected(<App />, { ...mockInitialState });
    documentBody.getByRole("heading", { name: "Lodgify Grouped Tasks" });
  });
  it("should render progress bar section", () => {
    const documentBody = renderConnected(<App />, { ...mockInitialState });
    documentBody.getByTestId("progressBarTestId");
  });
  it("should render groups section", () => {
    const documentBody = renderConnected(<App />, { ...mockInitialState });
    documentBody.getByTestId("groupsTestId");
  });
  it("should render 3 groups ", () => {
    const documentBody = renderConnected(<App />, { ...mockInitialState });
    const groups = documentBody.queryAllByTestId("groupHeaderTestId_", {
      exact: false,
    });
    expect(groups).toHaveLength(3);
  });
  it("should render 'Loading tasks' text when request is PENDING", () => {
    const documentBody = renderConnected(<App />, {
      tasks: { ...mockInitialState.tasks, loading: "PENDING" },
    });
    const loadingText = documentBody.queryByText("Loading Tasks");
    const progressSection = documentBody.queryByTestId("progressBarTestId");
    const groupsSection = documentBody.queryByTestId("groupsTestId");
    expect(loadingText).toBeInTheDocument();
    expect(groupsSection).not.toBeInTheDocument();
    expect(progressSection).not.toBeInTheDocument();
    //
  });
  it("should render 'Something went wrong!' text when request is REJECTED", () => {
    const documentBody = renderConnected(<App />, {
      tasks: { ...mockInitialState.tasks, loading: "REJECTED" },
    });
    const loadingText = documentBody.queryByText("Something went wrong!");
    const progressSection = documentBody.queryByTestId("progressBarTestId");
    const groupsSection = documentBody.queryByTestId("groupsTestId");
    expect(loadingText).toBeInTheDocument();
    expect(groupsSection).not.toBeInTheDocument();
    expect(progressSection).not.toBeInTheDocument();
  });
  it("should render 'No data to show' text when request is SUCCEEDED and no data", () => {
    const documentBody = renderConnected(<App />, {
      tasks: { loading: "SUCCEEDED", data: [] },
    });
    const loadingText = documentBody.queryByText("No data to show");
    expect(loadingText).toBeInTheDocument();
  });
});
