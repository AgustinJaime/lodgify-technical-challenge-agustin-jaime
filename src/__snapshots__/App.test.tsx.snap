// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> should render without issues 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app-container"
      >
        <header
          class="app-header"
        >
          <h1>
            Lodgify Grouped Tasks
          </h1>
        </header>
        <main>
          <section
            data-testid="progressBarTestId"
          >
            <div
              class="progress-bar-ctn"
            >
              <div
                class="progress-bar-base"
              >
                <div
                  class="progress-bar-inner"
                  style="padding: 0.5rem 0.5rem; width: 10%; transition: width 150ms ease;"
                >
                  <p>
                    10%
                  </p>
                </div>
              </div>
            </div>
          </section>
          <section
            class="groups-section"
            data-testid="groupsTestId"
          >
            <div
              class="group-ctn"
            >
              <div
                class="group-header"
                data-testid="groupHeaderTestId_0"
              >
                <div
                  class="group-header-section"
                >
                  <svg
                    class="list-i"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z"
                    />
                  </svg>
                  <p
                    class=""
                  >
                    Group 1
                  </p>
                </div>
                <div
                  class="group-header-section"
                >
                  <p>
                    Show
                  </p>
                  <svg
                    class="arrow-i "
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="group-ctn"
            >
              <div
                class="group-header"
                data-testid="groupHeaderTestId_1"
              >
                <div
                  class="group-header-section"
                >
                  <svg
                    class="list-i"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z"
                    />
                  </svg>
                  <p
                    class=""
                  >
                    Group 2
                  </p>
                </div>
                <div
                  class="group-header-section"
                >
                  <p>
                    Show
                  </p>
                  <svg
                    class="arrow-i "
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="group-ctn"
            >
              <div
                class="group-header"
                data-testid="groupHeaderTestId_2"
              >
                <div
                  class="group-header-section"
                >
                  <svg
                    class="list-i"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z"
                    />
                  </svg>
                  <p
                    class=""
                  >
                    Group 3
                  </p>
                </div>
                <div
                  class="group-header-section"
                >
                  <p>
                    Show
                  </p>
                  <svg
                    class="arrow-i "
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 256 256"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </section>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app-container"
    >
      <header
        class="app-header"
      >
        <h1>
          Lodgify Grouped Tasks
        </h1>
      </header>
      <main>
        <section
          data-testid="progressBarTestId"
        >
          <div
            class="progress-bar-ctn"
          >
            <div
              class="progress-bar-base"
            >
              <div
                class="progress-bar-inner"
                style="padding: 0.5rem 0.5rem; width: 10%; transition: width 150ms ease;"
              >
                <p>
                  10%
                </p>
              </div>
            </div>
          </div>
        </section>
        <section
          class="groups-section"
          data-testid="groupsTestId"
        >
          <div
            class="group-ctn"
          >
            <div
              class="group-header"
              data-testid="groupHeaderTestId_0"
            >
              <div
                class="group-header-section"
              >
                <svg
                  class="list-i"
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z"
                  />
                </svg>
                <p
                  class=""
                >
                  Group 1
                </p>
              </div>
              <div
                class="group-header-section"
              >
                <p>
                  Show
                </p>
                <svg
                  class="arrow-i "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="group-ctn"
          >
            <div
              class="group-header"
              data-testid="groupHeaderTestId_1"
            >
              <div
                class="group-header-section"
              >
                <svg
                  class="list-i"
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z"
                  />
                </svg>
                <p
                  class=""
                >
                  Group 2
                </p>
              </div>
              <div
                class="group-header-section"
              >
                <p>
                  Show
                </p>
                <svg
                  class="arrow-i "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="group-ctn"
          >
            <div
              class="group-header"
              data-testid="groupHeaderTestId_2"
            >
              <div
                class="group-header-section"
              >
                <svg
                  class="list-i"
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z"
                  />
                </svg>
                <p
                  class=""
                >
                  Group 3
                </p>
              </div>
              <div
                class="group-header-section"
              >
                <p>
                  Show
                </p>
                <svg
                  class="arrow-i "
                  fill="currentColor"
                  height="1em"
                  viewBox="0 0 256 256"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M181.66,133.66l-80,80a8,8,0,0,1-11.32-11.32L164.69,128,90.34,53.66a8,8,0,0,1,11.32-11.32l80,80A8,8,0,0,1,181.66,133.66Z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </section>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
